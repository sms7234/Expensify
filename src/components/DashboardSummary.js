import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses'

export class DashboardSummary extends React.Component {
  constructor(props){
    super(props);
    this.state={
      expenses: this.sumItems(this.filterExpenses(this.props.expenses)),
      income: this.sumItems(this.filterIncome(this.props.expenses))
    };
  }
  filterIncome = (arr) => {
    return arr.filter((item) => item.category==='Income')
  };
  filterExpenses = (arr) => {
    return arr.filter((item)=> item.category!=="Income")
  };
  sumItems = (arr) => {
    let holder = 0;
    if(arr!==[]){
      arr.forEach((item) => {
        holder = holder + item.amount
      })
    }
    return holder/100
  };


  render(){
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title"> Budgeting Dashboard </h1>
            <div className="content-container">
              <h2 className="page-header__subtitle">In the timeframe, specified below, you have earned <span> {numeral(this.state.expenses).format('$0,0.00')} </span> and spent <span> {numeral(this.state.income).format('$0,0.00')} </span></h2>
              <h2 className="page-header__subtitle">Remianing Funds: <span>{numeral(this.state.income - this.state.expenses).format('$0,0.00')}</span></h2>

            </div>
          </div>
        </div>
      </div>
    )
  }
}

const mapStateToProps=(state)=>{
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(DashboardSummary);
