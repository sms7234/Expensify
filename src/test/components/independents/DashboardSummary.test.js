import React from 'react';
import {shallow} from 'enzyme';
import DashboardSummary from '../../../components/independents/DashboardSummary';

test('should render the component with no data input', () => {
  const wrapper = shallow(<DashboardSummary />);
  expect(wrapper).toMatchSnapshot();
})

test('should render component with specified input', () => {
  const wrapper = shallow(<DashboardSummary income={2400} expenses={1500} />)
  expect(wrapper.find('span').at(0).text()).toBe(' $2,400.00 ');
  expect(wrapper.find('span').at(1).text()).toBe(' $1,500.00 ');
  expect(wrapper.find('span').at(2).text()).toBe('$900.00');
  expect(wrapper).toMatchSnapshot();
})
