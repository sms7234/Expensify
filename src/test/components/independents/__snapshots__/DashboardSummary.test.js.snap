// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component with specified input 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
         Budgeting Dashboard 
      </h1>
      <div
        className="content-container"
      >
        <h2
          className="page-header__subtitle"
        >
          In the timeframe, specified below, you have earned 
          <span>
             
            $2,400.00
             
          </span>
           and spent 
          <span>
             
            $1,500.00
             
          </span>
        </h2>
        <h2
          className="page-header__subtitle"
        >
          Remianing Funds: 
          <span>
            $900.00
          </span>
        </h2>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render the component with no data input 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
         Budgeting Dashboard 
      </h1>
      <div
        className="content-container"
      >
        <h2
          className="page-header__subtitle"
        >
          In the timeframe, specified below, you have earned 
          <span>
             
            $0.00
             
          </span>
           and spent 
          <span>
             
            $0.00
             
          </span>
        </h2>
        <h2
          className="page-header__subtitle"
        >
          Remianing Funds: 
          <span>
            $0.00
          </span>
        </h2>
      </div>
    </div>
  </div>
</div>
`;
